pkg/gather/clusterconfig/0_gatherer.go

```diff
@@ -1,25 +1,68 @@
 package clusterconfig
 
 import (
+	"time"
 	"context"
-	"sync"
+	"fmt"
+	"reflect"
+	"runtime"
+	"sort"
+	"strings"
 
 	"k8s.io/client-go/rest"
+	"k8s.io/klog"
 
-	configv1 "github.com/openshift/api/config/v1"
 	_ "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
 
 	"github.com/openshift/insights-operator/pkg/record"
 )
 
+type gatherStatusReport struct {
+	Name    string        `json:"name"`
+	Elapsed time.Duration `json:"elapsed"`
+	Report  int           `json:"report"`
+	Errors  []error       `json:"errors"`
+}
+
 // Gatherer is a driving instance invoking collection of data
 type Gatherer struct {
 	ctx                     context.Context
 	gatherKubeConfig        *rest.Config
 	gatherProtoKubeConfig   *rest.Config
 	metricsGatherKubeConfig *rest.Config
-	lock                    sync.Mutex
-	lastVersion             *configv1.ClusterVersion
+}
+
+type gatherFunction func(g *Gatherer) ([]record.Record, []error)
+
+const gatherAll = "ALL"
+
+var gatherFunctions = map[string]gatherFunction{
+	"pdbs": GatherPodDisruptionBudgets,
```

Comment:
maybe it would be easier if ids would match function names(e.g. gather_pod_distribution_budgets or pod_distribution_budgets)

Reply:
yeah, thought about that, currently I used either the name of the go file or the name of the folder inside the archive. (I've chosen based on personal preference :D )

how I think we should do it is have the ids match the name of the resource the could used in `oc get <resource_name>` (I think this is true for `pdbs`)